lib/web/apiserver.go

```diff
@@ -471,13 +472,23 @@ func NewHandler(cfg Config, opts ...HandlerOption) (*APIHandler, error) {
 
 			httplib.SetNoCacheHeaders(w.Header())
 
+			// /web/cluster/:clusterId/desktops/:desktopName/:username
+			// is a desktop session, which uses WASM and therefore needs a different CSP.
+			// We use a regex here to match that path in order to determine which CSP
+			// to set.
+			pattern := `^/web/cluster/[^/]+/desktops/[^/]+/[^/]+$`
+			re := regexp.MustCompile(pattern)
```

Comment:
Can you move this out so that we're not compiling the same regexp multiple times?

Reply:
eefd43f8da2a22c969477fb1373bd2f1e7bcaa11